File: tyre_pulse_flutter/lib/features/accidents/domain/accident_case_vocab.dart
/// The vocabulary of the owner's 10 accident mock screens, mirrored
/// byte-for-label from `src/lib/accidentCaseVocab.js` on the web. Change
/// BOTH together. Tokens are the live Supabase CHECK tokens; labels are what
/// the mocks print. Nothing here is a person's name or a country rule.
library;

import 'package:flutter/foundation.dart' show immutable;

@immutable
final class VocabItem {
  const VocabItem(
    this.key,
    this.label, {
    this.required = false,
    this.countable = false,
  });
  final String key;
  final String label;
  final bool required;
  final bool countable;
}

@immutable
final class NumberedStep {
  const NumberedStep(this.n, this.key, this.label, {this.owner = ''});
  final int n;
  final String key;
  final String label;
  final String owner;
}

/// Report wizard - "Step N of 7".
const List<NumberedStep> reportWizardSteps = <NumberedStep>[
  NumberedStep(1, 'identify_asset', 'Identify asset'),
  NumberedStep(2, 'incident', 'Incident details'),
  NumberedStep(3, 'people_authority', 'People and authority'),
  NumberedStep(4, 'damage', 'Mark damage'),
  NumberedStep(5, 'evidence', 'Evidence'),
  NumberedStep(6, 'documents', 'Documents'),
  NumberedStep(7, 'review', 'Review and submit'),
];

/// Case flow - "Workstream N of 7". Keys are the accident_case_workstreams
/// workstream_key values; damage_map and timeline are presentation-only.
const List<NumberedStep> caseFlow = <NumberedStep>[
  NumberedStep(1, 'fleet_validation', 'Fleet validation', owner: 'Fleet'),
  NumberedStep(2, 'assessment', 'Workshop assessment', owner: 'Workshop'),
  NumberedStep(3, 'insurance', 'Insurance / Claims', owner: 'Insurance'),
  NumberedStep(4, 'liability', 'Responsibility and payment', owner: 'Fleet'),
  NumberedStep(5, 'damage_map', 'Damage mapping', owner: 'Workshop'),
  NumberedStep(6, 'handover', 'Dispatch and handover', owner: 'Fleet'),
  NumberedStep(7, 'timeline', 'Timeline and closure', owner: 'Command Center'),
];

NumberedStep? caseFlowStep(String key) {
  for (final NumberedStep s in caseFlow) {
    if (s.key == key) return s;
  }
  return null;
}

String caseFlowLabel(String key) {
  final NumberedStep? s = caseFlowStep(key);
  return s == null ? '' : 'Workstream ${s.n} of ${caseFlow.length}';
}

/// M3 Who was at fault (accident_liability_assessments.liability_type).
@immutable
final class FaultTile {
  const FaultTile(this.key, this.label, {this.ourPct, this.otherPct});
  final String key;
  final String label;
  final int? ourPct;
  final int? otherPct;
}

const List<FaultTile> faultTiles = <FaultTile>[
  FaultTile('our_driver_full', 'Our driver / GCC', ourPct: 100, otherPct: 0),
  FaultTile('third_party_full', 'Other party', ourPct: 0, otherPct: 100),
  FaultTile('shared', 'Shared fault', ourPct: 50, otherPct: 50),
  FaultTile('under_investigation', 'Under investigation'),
  FaultTile('not_applicable', 'Not applicable'),
];

/// Mirrors the web `faultStatusFor`: an unset percentage is "not decided",
/// never Non-faulty.
String faultStatusFor(String? liabilityType, num? ourPct) {
  if (liabilityType == 'under_investigation') return 'Under review';
  if (liabilityType == 'not_applicable') return 'Not applicable';
  if (ourPct == null) return '';
  return ourPct > 0 ? 'Faulty' : 'Non-faulty';
}

/// M3 Who will pay (accident_liability_assessments.payer).
const List<VocabItem> payerTiles = <VocabItem>[
  VocabItem('other_party_insurance', 'Other party insurance'),
  VocabItem('our_insurance', 'Our insurance'),
  VocabItem('company', 'GCC / company'),
  VocabItem('driver_recovery', 'Driver recovery'),
  VocabItem('warranty', 'Warranty'),
  VocabItem('pending', 'Pending decision'),
];

String payerLabel(String? key) {
  for (final VocabItem p in payerTiles) {
    if (p.key == key) return p.label;
  }
  return '';
}

bool recoveryRequiredFor(String? payer) => const <String>{
      'other_party_insurance',
      'driver_recovery',
      'warranty',
    }.contains(payer);

/// M3 Third-party and authority rows (value | recorded by | verification).
const List<VocabItem> authorityRows = <VocabItem>[
  VocabItem('third_party_plate', 'Third-party plate'),
  VocabItem('third_party_driver', 'Third-party driver'),
  VocabItem('third_party_phone', 'Contact phone'),
  VocabItem('police_report_no', 'Police report no.'),
  VocabItem('najm_report', 'Najm report'),
  VocabItem('taqdeer_required', 'Taqdeer required'),
  VocabItem('taqdeer_no', 'Taqdeer no.'),
];

const List<String> verificationStates = <String>[
  'verified',
  'pending',
  'missing',
];

/// M3 Responsibility documents (accident_evidence.requirement_key).
const List<VocabItem> responsibilityDocs = <VocabItem>[
  VocabItem('police_accident_report', 'Police Accident Report', required: true),
  VocabItem('najm_report', 'Najm Report', required: true),
  VocabItem('taqdeer_assessment', 'Taqdeer Assessment', required: true),
  VocabItem(
    'third_party_registration_card',
    'Third-party Registration Card',
    required: true,
  ),
  VocabItem(
    'third_party_insurance_policy',
    'Third-party Insurance Policy',
    required: true,
  ),
  VocabItem('driver_licence', 'Driver Licence', required: true),
  VocabItem('company_letter_undertaking', 'Company Letter / Undertaking'),
];

/// M4 Claim document package (accident_evidence.requirement_key,
/// workstream insurance).
const List<VocabItem> claimPackageDocs = <VocabItem>[
  VocabItem('accident_report_pdf', 'Accident report PDF', required: true),
  VocabItem('fleet_validation', 'Fleet validation', required: true),
  VocabItem(
    'workshop_assessment_pdf',
    'Workshop assessment PDF',
    required: true,
  ),
  VocabItem(
    'damage_photographs',
    'Damage photographs',
    required: true,
    countable: true,
  ),
  VocabItem('police_najm_report', 'Police / Najm report', required: true),
  VocabItem('vehicle_registration', 'Vehicle registration', required: true),
  VocabItem('driving_licence', 'Driving licence', required: true),
  VocabItem('policy_document', 'Policy document', required: true),
];

/// M5 Repair assessment.
const List<VocabItem> repairRouteTiles = <VocabItem>[
  VocabItem('internal', 'Internal workshop'),
  VocabItem('external', 'External workshop'),
  VocabItem('on_site', 'On-site repair'),
];

const List<VocabItem> damageActions = <VocabItem>[
  VocabItem('repair', 'Repair'),
  VocabItem('replace', 'Replace'),
  VocabItem('structural_review', 'Replace / structural review'),
  VocabItem('monitor', 'Monitor'),
];

const List<VocabItem> quotationStates = <VocabItem>[
  VocabItem('not_requested', 'Not requested'),
  VocabItem('requested', 'Requested'),
  VocabItem('received', 'Received'),
  VocabItem('approved', 'Approved'),
  VocabItem('rejected', 'Rejected'),
];

const List<VocabItem> assessmentAttachments = <VocabItem>[
  VocabItem('assessment_report_pdf', 'Assessment report PDF', required: true),
  VocabItem('vendor_quotation', 'Vendor quotation', required: true),
  VocabItem(
    'damage_photos',
    'Damage photos',
    required: true,
    countable: true,
  ),
  VocabItem('recovery_request', 'Recovery request'),
];

/// The one attachment that gates "Submit assessment and route" for an
/// external repair route.
const String submitGatingAttachment = 'vendor_quotation';

/// M6 Fleet validation checklist (accident_fleet_validation_items.item_key).
const List<VocabItem> fleetValidationItems = <VocabItem>[
  VocabItem('asset_driver_confirmed', 'Asset and driver confirmed'),
  VocabItem('incident_facts_confirmed', 'Incident facts confirmed'),
  VocabItem('damage_map_reviewed', 'Damage map reviewed'),
  VocabItem('required_photographs', 'Required photographs', countable: true),
  VocabItem(
    'police_najm_documents',
    'Police / Najm documents',
    countable: true,
  ),
  VocabItem(
    'workshop_assessment_requested',
    'Workshop assessment requested',
  ),
];

const List<String> checkStates = <String>[
  'pending',
  'done',
  'attention',
  'not_applicable',
];

/// M2 Dispatch and handover.
const List<NumberedStep> dispatchStepper = <NumberedStep>[
  NumberedStep(1, 'dispatched', 'Dispatched'),
  NumberedStep(2, 'arrived', 'Arrived'),
  NumberedStep(3, 'signed_acceptance', 'Signed acceptance'),
  NumberedStep(4, 'vendor_assessment', 'Vendor assessment / quotation starts'),
];

const List<VocabItem> dispatchLiveStates = <VocabItem>[
  VocabItem('preparing', 'Preparing'),
  VocabItem('in_transit', 'In transit'),
  VocabItem('arrived', 'Arrived'),
  VocabItem('accepted', 'Accepted'),
];

/// Vendor receipt fields the mock marks with an asterisk (accident_dispatches).
const List<String> receiptRequired = <String>[
  'arrived_at',
  'received_by_name',
  'received_by_designation',
  'receiving_photos',
  'handover_paper_ref',
  'receiver_signature',
  'custody_accepted',
];

/// M8/M9/M10 Damage mapping.
const List<VocabItem> damageTypes = <VocabItem>[
  VocabItem('dent', 'Dent'),
  VocabItem('scratch', 'Scratch'),
  VocabItem('cracked', 'Cracked'),
  VocabItem('broken', 'Broken'),
  VocabItem('missing', 'Missing'),
  VocabItem('bent', 'Bent'),
  VocabItem('other', 'Other'),
];

String canonDamageType(String? raw) {
  final String k = (raw ?? '').trim().toLowerCase();
  if (k.isEmpty) return '';
  if (k == 'crack' || k == 'cracked') return 'cracked';
  if (k == 'dented') return 'dent';
  for (final VocabItem d in damageTypes) {
    if (d.key == k) return k;
  }
  return 'other';
}

/// Levels keep the stored tokens; 'severe' is labelled Major.
const List<VocabItem> damageLevels = <VocabItem>[
  VocabItem('minor', 'Minor'),
  VocabItem('moderate', 'Moderate'),
  VocabItem('severe', 'Major'),
];

const int damageNoteMax = 200;

/// View chips per vehicle family in mock order. Keys match the web
/// FAMILY_VIEW_ORDER; 'front_left' and 'top' are the additions.
const Map<String, List<String>> familyViewOrder = <String, List<String>>{
  'bus': <String>['left', 'front_left', 'front', 'right', 'rear', 'top'],
  'concrete_pump': <String>['top', 'left', 'right', 'front', 'rear'],
  'pickup': <String>['left', 'right', 'front', 'rear', 'top'],
  'generic': <String>['left', 'right', 'front', 'rear', 'top'],
};

const Map<String, String> viewLabels = <String, String>{
  'left': 'Left',
  'front_left': 'Front-left',
  'front': 'Front',
  'right': 'Right',
  'rear': 'Rear',
  'top': 'Top',
  'overview': 'Overview',
};

/// Named notify roles on M4/M5/M6 (roles, never people).
@immutable
final class NotifyRole {
  const NotifyRole(
    this.key,
    this.label,
    this.roles, {
    this.visibilityOnly = false,
  });
  final String key;
  final String label;
  final List<String> roles;
  final bool visibilityOnly;
}

const List<NotifyRole> notifyRoles = <NotifyRole>[
  NotifyRole('fleet', 'Fleet', <String>['Fleet Supervisor', 'Manager']),
  NotifyRole(
    'workshop',
    'Workshop',
    <String>['Workshop Supervisor', 'Workshop Maintenance Area Manager'],
  ),
  NotifyRole('insurance', 'Insurance', <String>['Insurance Officer']),
  NotifyRole(
    'command_center',
    'Command Center',
    <String>['Data Monitor Officer'],
  ),
  NotifyRole(
    'pmv_manager',
    'PMV Manager',
    <String>['PMV Manager'],
    visibilityOnly: true,
  ),
];

/// M1 Timeline.
const List<String> timelineTabs = <String>[
  'Timeline',
  'Notifications',
  'Participants',
];
const List<String> timelineFilters = <String>[
  'all',
  'actions',
  'documents',
  'sla',
  'emails',
];

File: src/lib/accidentCaseVocab.js
/**
 * accidentCaseVocab.js - the vocabulary the owner's accident mock screens use,
 * in ONE place so the web case tabs and the Flutter case screens cannot drift.
 *
 * Source: the 10 mock screens shared on 2026-09-16 (case timeline, dispatch and
 * handover, responsibility and payment, register claim, repair assessment,
 * fleet validation, identify asset, and three damage-map screens). Every list
 * below is transcribed from those screens. Do NOT re-declare any of these in a
 * component; import from here. accidentVocab.js keeps the OLDER register
 * vocabularies (severity/status/type tokens) and is untouched.
 *
 * DB tokens are snake_case; labels are what the mock prints. Where a token
 * already exists in the live schema (liability_type, recommended_route,
 * severity) the mock label is mapped onto the EXISTING token rather than a new
 * one, so nothing already stored is invalidated.
 */

// ── Report wizard (mobile) - "Step N of 7" ───────────────────────────────────
export const REPORT_WIZARD_STEPS = [
  { key: 'identify_asset',   n: 1, label: 'Identify asset' },
  { key: 'incident',         n: 2, label: 'Incident details' },
  { key: 'people_authority', n: 3, label: 'People and authority' },
  { key: 'damage',           n: 4, label: 'Mark damage' },
  { key: 'evidence',         n: 5, label: 'Evidence' },
  { key: 'documents',        n: 6, label: 'Documents' },
  { key: 'review',           n: 7, label: 'Review and submit' },
]

// ── Case flow (both apps) - "Workstream N of 7" ──────────────────────────────
// Maps the mock screen order onto the accidentCase.js WORKSTREAMS keys so the
// same progress engine keeps driving completion; the mock numbering is only a
// presentation order.
export const CASE_FLOW = [
  { n: 1, key: 'fleet_validation', label: 'Fleet validation',          owner: 'Fleet' },
  { n: 2, key: 'assessment',       label: 'Workshop assessment',       owner: 'Workshop' },
  { n: 3, key: 'insurance',        label: 'Insurance / Claims',        owner: 'Insurance' },
  { n: 4, key: 'liability',        label: 'Responsibility and payment', owner: 'Fleet' },
  { n: 5, key: 'damage_map',       label: 'Damage mapping',            owner: 'Workshop' },
  { n: 6, key: 'handover',         label: 'Dispatch and handover',     owner: 'Fleet' },
  { n: 7, key: 'timeline',         label: 'Timeline and closure',      owner: 'Command Center' },
]
export const caseFlowStep = (key) => CASE_FLOW.find((s) => s.key === key) || null
export const caseFlowLabel = (key) => {
  const s = caseFlowStep(key)
  return s ? `Workstream ${s.n} of ${CASE_FLOW.length}` : ''
}

// ── M3 Who was at fault (maps onto accident_liability_assessments.liability_type)
export const FAULT_TILES = [
  { key: 'our_driver_full',     label: 'Our driver / GCC',    ourPct: 100, otherPct: 0 },
  { key: 'third_party_full',    label: 'Other party',         ourPct: 0,   otherPct: 100 },
  { key: 'shared',              label: 'Shared fault',        ourPct: 50,  otherPct: 50 },
  { key: 'under_investigation', label: 'Under investigation', ourPct: null, otherPct: null },
  { key: 'not_applicable',      label: 'Not applicable',      ourPct: null, otherPct: null },
]
export const faultStatusFor = (liabilityType, ourPct) => {
  if (liabilityType === 'under_investigation') return 'Under review'
  if (liabilityType === 'not_applicable') return 'Not applicable'
  // Number(null) is 0 and 0 is finite - an unset percentage must read as
  // "not decided", never as Non-faulty. Check blankness before coercing.
  if (ourPct === null || ourPct === undefined || String(ourPct).trim() === '') return ''
  const n = Number(ourPct)
  if (!Number.isFinite(n)) return ''
  return n > 0 ? 'Faulty' : 'Non-faulty'
}

// ── M3 Who will pay (accident_liability_assessments.payer) ───────────────────
export const PAYER_TILES = [
  { key: 'other_party_insurance', label: 'Other party insurance' },
  { key: 'our_insurance',         label: 'Our insurance' },
  { key: 'company',               label: 'GCC / company' },
  { key: 'driver_recovery',       label: 'Driver recovery' },
  { key: 'warranty',              label: 'Warranty' },
  { key: 'pending',               label: 'Pending decision' },
]
export const payerLabel = (k) => PAYER_TILES.find((p) => p.key === k)?.label || ''
/** Recovery is "required" when someone other than us pays. */
export const recoveryRequiredFor = (payer) => ['other_party_insurance', 'driver_recovery', 'warranty'].includes(payer)

// ── M3 Third-party and authority rows (each shows value, recorded by, verification)
export const AUTHORITY_ROWS = [
  { key: 'third_party_plate',  label: 'Third-party plate' },
  { key: 'third_party_driver', label: 'Third-party driver' },
  { key: 'third_party_phone',  label: 'Contact phone' },
  { key: 'police_report_no',   label: 'Police report no.' },
  { key: 'najm_report',        label: 'Najm report' },
  { key: 'taqdeer_required',   label: 'Taqdeer required' },
  { key: 'taqdeer_no',         label: 'Taqdeer no.' },
]
export const VERIFICATION_STATES = ['verified', 'pending', 'missing']

// ── M3 Responsibility documents (accident_evidence.requirement_key) ─────────
export const RESPONSIBILITY_DOCS = [
  { key: 'police_accident_report',       label: 'Police Accident Report',        required: true },
  { key: 'najm_report',                  label: 'Najm Report',                   required: true },
  { key: 'taqdeer_assessment',           label: 'Taqdeer Assessment',            required: true },
  { key: 'third_party_registration_card', label: 'Third-party Registration Card', required: true },
  { key: 'third_party_insurance_policy', label: 'Third-party Insurance Policy',  required: true },
  { key: 'driver_licence',               label: 'Driver Licence',                required: true },
  { key: 'company_letter_undertaking',   label: 'Company Letter / Undertaking',  required: false },
]

// ── M4 Claim document package (accident_evidence.requirement_key, workstream insurance)
export const CLAIM_PACKAGE_DOCS = [
  { key: 'accident_report_pdf',      label: 'Accident report PDF',      required: true },
  { key: 'fleet_validation',         label: 'Fleet validation',         required: true },
  { key: 'workshop_assessment_pdf',  label: 'Workshop assessment PDF',  required: true },
  { key: 'damage_photographs',       label: 'Damage photographs',       required: true, countable: true },
  { key: 'police_najm_report',       label: 'Police / Najm report',     required: true },
  { key: 'vehicle_registration',     label: 'Vehicle registration',     required: true },
  { key: 'driving_licence',          label: 'Driving licence',          required: true },
  { key: 'policy_document',          label: 'Policy document',          required: true },
]

// ── M5 Repair assessment ─────────────────────────────────────────────────────
// Maps the three mock tiles onto the EXISTING recommended_route / repair_route
// CHECK tokens. 'on_site' is added by the parity migration.
export const REPAIR_ROUTE_TILES = [
  { key: 'internal', label: 'Internal workshop' },
  { key: 'external', label: 'External workshop' },
  { key: 'on_site',  label: 'On-site repair' },
]
export const DAMAGE_ACTIONS = [
  { key: 'repair',            label: 'Repair' },
  { key: 'replace',           label: 'Replace' },
  { key: 'structural_review', label: 'Replace / structural review' },
  { key: 'monitor',           label: 'Monitor' },
]
export const PARTS_AVAILABILITY = [
  { key: 'available',     label: 'Available' },
  { key: 'special_order', label: 'Special order' },
  { key: 'unknown',       label: 'Not checked' },
]
export const QUOTATION_STATES = [
  { key: 'not_requested', label: 'Not requested' },
  { key: 'requested',     label: 'Requested' },
  { key: 'received',      label: 'Received' },
  { key: 'approved',      label: 'Approved' },
  { key: 'rejected',      label: 'Rejected' },
]
export const ASSESSMENT_ATTACHMENTS = [
  { key: 'assessment_report_pdf', label: 'Assessment report PDF', required: true },
  { key: 'vendor_quotation',      label: 'Vendor quotation',      required: true, gatesSubmit: true },
  { key: 'damage_photos',         label: 'Damage photos',         required: true, countable: true },
  { key: 'recovery_request',      label: 'Recovery request',      required: false },
]

// ── M6 Fleet validation checklist (accident_fleet_validation_items.item_key) ─
export const FLEET_VALIDATION_ITEMS = [
  { key: 'asset_driver_confirmed',       label: 'Asset and driver confirmed' },
  { key: 'incident_facts_confirmed',     label: 'Incident facts confirmed' },
  { key: 'damage_map_reviewed',          label: 'Damage map reviewed' },
  { key: 'required_photographs',         label: 'Required photographs', countable: true },
  { key: 'police_najm_documents',        label: 'Police / Najm documents', countable: true },
  { key: 'workshop_assessment_requested', label: 'Workshop assessment requested' },
]
export const CHECK_STATES = ['pending', 'done', 'attention', 'not_applicable']

// ── M2 Dispatch and handover ─────────────────────────────────────────────────
export const DISPATCH_STEPPER = [
  { key: 'dispatched',        n: 1, label: 'Dispatched' },
  { key: 'arrived',           n: 2, label: 'Arrived' },
  { key: 'signed_acceptance', n: 3, label: 'Signed acceptance' },
  { key: 'vendor_assessment', n: 4, label: 'Vendor assessment / quotation starts' },
]
export const DISPATCH_LIVE_STATES = [
  { key: 'preparing',  label: 'Preparing' },
  { key: 'in_transit', label: 'In transit' },
  { key: 'arrived',    label: 'Arrived' },
  { key: 'accepted',   label: 'Accepted' },
]
/** Vendor receipt fields the mock marks with an asterisk. */
export const RECEIPT_REQUIRED = [
  'arrived_at', 'received_by_name', 'received_by_designation',
  'receiving_photos', 'handover_paper_ref', 'receiver_signature', 'custody_accepted',
]

// ── M8/M9/M10 Damage mapping ────────────────────────────────────────────────
export const DAMAGE_TYPES = [
  { key: 'dent',    label: 'Dent' },
  { key: 'scratch', label: 'Scratch' },
  { key: 'cracked', label: 'Cracked' },
  { key: 'broken',  label: 'Broken' },
  { key: 'missing', label: 'Missing' },
  { key: 'bent',    label: 'Bent' },
  { key: 'other',   label: 'Other' },
]
export const DAMAGE_TYPE_ALIAS = { crack: 'cracked', cracked: 'cracked', dented: 'dent' }
export const canonDamageType = (v) => {
  const k = String(v || '').trim().toLowerCase()
  return DAMAGE_TYPE_ALIAS[k] || (DAMAGE_TYPES.some((d) => d.key === k) ? k : k ? 'other' : '')
}
// Levels keep the EXISTING stored tokens (minor/moderate/severe); the mock
// label for 'severe' is Major.
export const DAMAGE_LEVELS = [
  { key: 'minor',    label: 'Minor' },
  { key: 'moderate', label: 'Moderate' },
  { key: 'severe',   label: 'Major' },
]
export const DAMAGE_NOTE_MAX = 200
/** View chips per vehicle family, in the order the mock shows them. */
export const FAMILY_VIEW_ORDER = {
  bus:           ['left', 'front_left', 'front', 'right', 'rear', 'top'],
  concrete_pump: ['top', 'left', 'right', 'front', 'rear'],
  pickup:        ['left', 'right', 'front', 'rear', 'top'],
  generic:       ['left', 'right', 'front', 'rear', 'top'],
}
export const VIEW_LABELS = {
  left: 'Left', front_left: 'Front-left', front: 'Front', right: 'Right', rear: 'Rear', top: 'Top', overview: 'Overview',
}

// ── Named notify roles used on M4/M5/M6 ─────────────────────────────────────
// Roles, not people: the mock shows names because a person holds the role.
export const NOTIFY_ROLES = [
  { key: 'fleet',          label: 'Fleet',          roles: ['Fleet Supervisor', 'Manager'] },
  { key: 'workshop',       label: 'Workshop',       roles: ['Workshop Supervisor', 'Workshop Maintenance Area Manager'] },
  { key: 'insurance',      label: 'Insurance',      roles: ['Insurance Officer'] },
  { key: 'command_center', label: 'Command Center', roles: ['Data Monitor Officer'] },
  { key: 'pmv_manager',    label: 'PMV Manager',    roles: ['PMV Manager'], visibilityOnly: true },
]

// ── M1 Timeline filters (already in caseTimelineFeed.FILTERS; mirrored here for Flutter parity)
export const TIMELINE_TABS = ['Timeline', 'Notifications', 'Participants']
export const TIMELINE_FILTERS = ['all', 'actions', 'documents', 'sla', 'emails']
